readalongs/align.py

```diff
@@ -426,47 +423,50 @@ def align_sequence(
         write_audio_to_file(audio_segment, save_temps + ".wav" + i_suffix)
 
     # Configure soundswallower for this sequence's dict and fsg
-    asr_config.set_string("-dict", dict_file.name)
-    asr_config.set_string("-fsg", fsg_file.name)
+    asr_config["dict"] = dict_file.name
+    asr_config["fsg"] = fsg_file.name
 
     ps = soundswallower.Decoder(asr_config)
     # Align this word sequence
     ps.start_utt()
     ps.process_raw(audio_segment.raw_data, no_search=False, full_utt=True)
     ps.end_utt()
 
-    return ps.seg()
+    return ps.seg
 
 
 def process_segmentation(
-    segmentation: Iterable[soundswallower.Segment],
+    segmentation: Iterable[soundswallower.Seg],
     curr_removed_segments: List[dict],
     noisewords: Set[str],
     frame_size: float,
     debug_aligner: Optional[bool] = False,
 ) -> List[Dict[str, Any]]:
     """Correct output alignments based on do-not-align segments."""
-    aligned_words = []
+    aligned_words: List[Dict[str, Any]] = []
     for word_seg in segmentation:
-        if word_seg.word in noisewords:
+        if word_seg.text in noisewords:
             continue
-        start = word_seg.start_frame * frame_size
-        end = (word_seg.end_frame + 1) * frame_size
-        # change to ms
-        start_ms = int(start * 1000)
-        end_ms = int(end * 1000)
+        start = word_seg.start
+        end = word_seg.start + word_seg.duration
+        # round to milliseconds to avoid imprecisions
+        start_ms = round(start * 1000)
+        end_ms = round(end * 1000)
+        # possibly adjust for removed sections
         if curr_removed_segments:
             start_ms += calculate_adjustment(start_ms, curr_removed_segments)
             end_ms += calculate_adjustment(end_ms, curr_removed_segments)
             start_ms, end_ms = correct_adjustments(
                 start_ms, end_ms, curr_removed_segments
             )
-            # change back to seconds to write to smil
-            start = start_ms / 1000
-            end = end_ms / 1000
-        aligned_words.append({"id": word_seg.word, "start": start, "end": end})
+        # change back to seconds
```

Comment:
Argh, this back and forth between seconds and ms is confusing! But that's not your doing, it was already like that. At least the variables that hold ms are called `_ms` so we can make sense of it.

Reply:
yes, the important thing is to ensure that everything is rounded to milliseconds consistently...